include/triton/common/triton_json.h

```diff
@@ -194,6 +200,31 @@ class TritonJson {
       return Parse(json.data(), json.size());
     }
 
+    // Helper function for Parse(const char* base, const size_t size) to handle
+    // errors. Return error message if parsing failed.
+    TRITONJSON_STATUSTYPE ParseErrorHandler(
+        const rapidjson::Document& document, const std::string& json)
+    {
+      if (document.HasParseError()) {
+        std::ostringstream error_stream;
+        error_stream << "failed to parse the request JSON buffer: "
+                     << GetParseError_En(document.GetParseError())
+                     << " at offset " << document.GetErrorOffset() << ".";
+
+        // Show part of the JSON to help debugging
+        const size_t preview_length = 100;
+        std::string json_preview = json.substr(0, preview_length);
+        if (json.size() > preview_length) {
+          json_preview += "...";
+        }
```

Comment:
Any strong reason to not just include the whole thing?

For example, can the error start at some point after the preview length?

```{... /* 100+ chars */, "my_invalid_json".123}```

Reply:
I wanted to avoid that in case the JSON might be very long and crashes the program. I can increase the preview length to the limit of your choice (maybe `std::numeric_limits<std::size_t>::max()`or print the whole thing. Any preference?

Reply:
That is a fair point that it could very well be an issue after the 100th character. A large limit could resolve that while balancing the risk of the JSON being too long and causing a crash.